### ISSUE TITLE
rails-assets.org is down

### ISSUE THREAD

**Issue by ypadlyak** [tenex/rails-assets#329]
Are there any workaround until it is up?

**Comment by bbraga** [comment]
rails-assets.tenex.tech is also down, I believe is the server that the domain is pointing to, which is my guess. 

This is why we need a mirror approach to hosting this, we cannot have one single entity doing it as any downtime will affect the projects credibility.

**Comment by kofronpi** [comment]
Using `mtr`, both rails-assets.org and rails-assets.tenex.tech  point to `digitalocean-ic-306497-nyk-b3.c.telia.net` then the connection is hanging, so I guess it's a configuration problem.

**Comment by gingerlime** [comment]
TL; DR - Sorry to sound negative. Wasn't aware of #291, but bumped into a couple of issues with rails-assets recently (#328 #326), and some availability issues in the past. So this is the last nail in the coffin for us...

I really think rails-assets is a great idea, and very well executed as a concept. Also I really appreciate the responsiveness of @hut8 so far, and genuinely wish the best success to this project. I would like to thank everyone involved in making it happen and hosting it.

Sadly, we can't rely on the project, and don't have the resources to support it to make it more stable, so we'll have to figure out a different solution (currently looking at rails-browserify and until then, we'll manually copy JS files into the asset pipeline, like in the good ol' days).

**Comment by ssaunier** [comment]
I think `rails-assets` is an awesome project ❤️ , I'm sure you guys will fix that downtime issue, in the meantime, I'll just `bundle package` stuff (cf https://github.com/rails-assets/rails-assets/issues/291#issuecomment-218106837)

Good luck fixing this outage.

**Comment by bbraga** [comment]
To everyone looking for an easy replacement:
https://github.com/rharriso/bower-rails

I just migrated all 35 of my assets into it in 5 mins. 

Good luck all!

**Comment by dreyks** [comment]
Just succeeded an install. Everything seems to be ok right now

**Comment by pinak1180** [comment]
yes it seems back

**Comment by joshjordan** [comment]
Copy/pasta from the other thread: We weren't taking any more traffic than usual last night, and syslog isn't showing anything useful. Looks like we went hard down at 3:41AM. Unfortunately, I wasn't awake until a few minutes after we recovered, so I'm desperate for more data. I'm following up with our hosting provider to get more info. Right now, we appear to be up and stable. I'm going to keep this issue open until we determine the root cause.

@bbraga I appreciate your offer to host a mirror! Please let me know what help you need.

I suspect a mirror will not help very much, because we won't have a good way to communicate the existence of a mirror in a scenario like this, and past evidence indicates that people will be just as annoyed or reluctant to make a temporary change in their Gemfile if the site is down. We may need some dynamic failover, but I can't answer that until we finish a post-mortem and figure out what actually happened.

@gingerlime makes sense to me. It isn't for everyone. Cheers.

**Comment by hut8** [comment]
Looks like the cause of this is that there was a hardware issue on the main rails-assets.org droplet. That's pretty hard to protect against, and as a part of resolving that issue, Digital Ocean rebooted it and afterwords all services came back up normally. That was a large amount of downtime and for that I apologize for the unavailability of the service. However, if a datacenter has some downtime there's not a whole lot we can do.

Going forward, I will look into deploying a second instance and throw up both behind a loadbalancer, but that's not going to be a trivial amount of work.

**Comment by bbraga** [comment]
@hut8 Hi, I have moved on from rails-assets, but I would like to offer some idea on how to prevent this in the future.

Since DigitalOcean is sponsoring the hosting for this project, wouldn't it be possible/sensible to have more than one droplet in a distributed fashion ? 

Example infrastructure: 
Droplet 1: load-balancer nginx US-EAST
Droplet 2: load-balancer nginx EU-WEST
Droplet 3: rails-assets US-EAST
Droplet 4: rails-assets US-WEST
Droplet 5: rails-assets EU-WEST

Add more droplets for more redundancy. 

I don't see how today's event could have not been mitigated with such an approach as, according to your post, it was an hardware/datacenter issue. 

Note: due to the critical nature of this project, deploying and bundling will not work, meaning that if this is not addressed, it will make production systems that depend on this to fail, which in turn will further jeopardize the projects credibility as well as anyone associated with it.

Best of luck to you guys in solving this.

**Comment by stefanahman** [comment]
It is down again.

**Comment by anonoz** [comment]
And it is down again.

**Comment by izikaj** [comment]
And it is up again

**Comment by joshjordan** [comment]
This issue is two years old and isn't likely to help others searching for answers. Please create a new one or participate in the active thread in the future.

**Comment by marek-mazur** [comment]
https://github.com/tenex/rails-assets/issues/453 ;-)
